core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -413,10 +413,13 @@ private[spark] class SparkSubmitArguments(args: Seq[String], env: Map[String, St
       case ("--help" | "-h") :: tail =>
         printUsageAndExit(0)
 
-      case ("--verbose" | "-v") :: tail =>
+      case ("--verbose") :: tail =>
         verbose = true
         parse(tail)
 
+      case ("--version" | "-v") :: tail =>
+        SparkSubmit.printVersionAndExit()
```

Comment:
Some binaries, like Java, don't exit just because the version is printed. Rather than push some flow control and exit from the argument parsing code, this could become another flag in the args object, that later causes version to print in the normal flow.